### DISCUSSION TITLE
Does the virtual table only still support read tables?

### DISCUSSION THREAD

**Discussion by iamwilhelm** [vrischmann/zig-sqlite#141] [General]
I read that in the [introductory post](https://rischmann.fr/blog/virtual-tables-with-zig-sqlite). However, it's been a little while, and based on what I know from reading the Sqlite documentation, enabling writing is being able to define functions like `xUpdate` and the like. But as far as I can tell, looking at the `sqlite3_module`, `xUpdate` and the rest of the struct function API is already defined. 

So I'm guessing virtual tables can also handle writes, if the appropriate callbacks are implemented?

**Comment by vrischmann** [comment]
It's not implemented, the callbacks are [set to null](https://github.com/vrischmann/zig-sqlite/blob/master/vtab.zig#L656-L666). 

You need to implement the callbacks but also design and implement the zig abstractions, which is going to take more time.
